src/mongocxx/private/collection.hpp

```diff
@@ -56,10 +55,10 @@ class collection::impl {
     }
 
     mongoc_collection_t* collection_t;
-    std::string database_name;
+    bsoncxx::string::view_or_value database_name;
```

Comment:
If you wanted to create many database objects with the same name, you might want to do it using views of one string object, to avoid making copies of it.  It's the same argument for why you might want to use views of a document, except the case for a std::string is weaker.

Specifically for collections, you're probably creating your collections using `db["coll_name"]`, and the `database_name` member here is built from the parent db's stored name.  Before, we copied that std::string into every new collection, now we don't need to.